### PR TITLE
add - unit tests

### PR DESCRIPTION
**Pull request by ilinzweilin**

### CODE REVIEW

File: src/test/mgr-unit.t.sol [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+        assertEq(vat.values_int("frob_dink"), -int(wad)); 
+        assertEq(vat.values_int("frob_dart"), 0); 
+    }
+
+    function draw(uint wad) public {
+        uint selfBalanceDAI = dai.balanceOf(self);
+
+        mgr.draw(wad);
+        
+        // chack DAI were minted
```
Comment (xmxanuel) [inline src/test/mgr-unit.t.sol]:
```suggestion
        // check DAI were minted
```

File: src/test/mgr-unit.t.sol [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+        assertEq(vat.values_int("frob_dart"), int(wad)); 
+    }
+
+    function wipe(uint wad) public {
+        uint selfBalanceDAI = dai.balanceOf(self);
+        uint mgrBalanceDAI = dai.balanceOf(mgr_);
+
+        mgr.wipe(wad);
+
+         // chack DAI were minted
```
Comment (xmxanuel) [inline src/test/mgr-unit.t.sol]:
```suggestion
         // check DAI were minted
```

File: src/mgr.sol [hunk 3]
```diff
@@ -70,7 +71,7 @@ interface RedeemLike {
 // not only DROP as an ERC20 balance in this contract, but also what's currently
 // undergoing redemption from the Tinlake pool.
 
-contract TinlakeManager is LibNote {
+contract TinlakeManager is LibNote, DSTest {
```
Comment (xmxanuel) [inline src/mgr.sol]:
```suggestion
contract TinlakeManager is LibNote {
```

File: src/mgr.sol [hunk 4]
```diff
@@ -18,6 +18,7 @@
 
 pragma solidity >=0.5.12;
 import "./lib.sol";
+import "ds-test/test.sol";
```
Comment (xmxanuel) [inline src/mgr.sol]:
```suggestion

```

File: src/test/mocks/auth.sol [hunk 5]
```diff
@@ -0,0 +1,27 @@
+// Copyright (C) 2020 Centrifuge
```
Comment (xmxanuel) [inline src/test/mocks/auth.sol:1]: not really a mock

File: src/test/mocks/vat.sol [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    function increaseTab(uint wad) external {
+        values_uint["tab"] = safeAdd(values_uint["tab"], wad);
+    }
+
+    function decreaseTab(uint wad) external {
+        values_uint["tab"] = safeSub(values_uint["tab"], wad);
+    }
+}
```
Comment (xmxanuel) [inline src/test/mocks/vat.sol]: the `values_uint` are not consistent with the rest.

I think it should be values_uint["funcname_variable"]`

File: src/test/mgr-unit.t.sol [hunk 7]
```diff
@@ -0,0 +1,454 @@
...
+        assertEq(seniorOperator.calls("redeemOrder"), 1);
+        assertEq(seniorOperator.values_uint("redeemOrder_wad"), wad);
+    }
+
+    function cage() public {
+        mgr.cage();
+        assert(!mgr.live());
+    }
+
+    function  changeOwner() public {
```
Comment (xmxanuel) [inline src/test/mgr-unit.t.sol]:
```suggestion
    function changeOwner() public {
```

**Review by xmxanuel — APPROVED** [review]
LGTM

File: src/mgr.sol [hunk 8]
```diff
@@ -208,7 +209,7 @@ contract TinlakeManager is LibNote {
 
     // --- Liquidation ---
     function tell() public note {
-        require(safe);
+        require(safe);
```
Comment (MrChico) [inline src/mgr.sol:212]:
```suggestion
        require(safe);
```

**Pull request merged**